src/compas_fab/backends/client_manager.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+import compas
+if compas.is_ironpython():
```

Comment:
This file I find the most problematic. As mentioned below, `contextlib2` is an open question about safe install to ironpython. And while I understand the joy of the fancy resource handling here, I don't think we have a planned usage that justifies it.

I still think that combinining clients into one new client is something that is not done ad-hoc but the end-user, but more of a decision by developers trying to add support for something new, in which case they will code a new class and use that instead.